task-log/task_log.go

```diff
@@ -0,0 +1,77 @@
+package taskLog
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/taskcluster/taskcluster-cli/apis"
+	"github.com/taskcluster/taskcluster-cli/extpoints"
+)
+
+type taskLog struct{}
+
+func init() {
+	extpoints.Register("task-log", taskLog{})
+}
+
+func (taskLog) ConfigOptions() map[string]extpoints.ConfigOption {
+	return nil
+}
+
+func (taskLog) Summary() string {
+	return "Outputs the logs for <taskID> as generated, and exits when completes."
+}
+
+func (taskLog) Usage() string {
+	usage := "Usage: taskcluster task-log <taskID>\n"
+	return usage
+}
+
+func (taskLog) Execute(context extpoints.Context) bool {
+	taskID := context.Arguments["<taskID>"].(string)
+
+	// Get route from services.go
+	route := apis.Services["Queue"].BaseURL
+	for _, entry := range apis.Services["Queue"].Entries {
+		if entry.Name == "getLatestArtifact" {
+			route += entry.Route
+			break
+		}
+	}
+
+	route = strings.Replace(route, "<taskId>", taskID, 1)
+	route = strings.Replace(route, "<name>", "public/logs/live_backing.log", 1)
```

Comment:
For running tasks using `live.log`,  the get request will hang until the task is completed. I chose to use `live_backing.log` for this pr because we don't have a proper way to deal with live tasks at the moment. So in this case using `live_backing.log` will at least give us an error for running tasks.